lms/djangoapps/instructor_task/api.py

```diff
@@ -433,7 +435,16 @@ def generate_certificates_for_all_students(request, course_key):   # pylint: dis
     task_input = {}
     task_key = ""
 
-    return submit_task(request, task_type, task_class, course_key, task_input, task_key)
+    instructor_task = submit_task(request, task_type, task_class, course_key, task_input, task_key)
```

Comment:
It looks to me we have duplicate methods `generate_certificates_for_students` and `generate_certificates_for_all_students` and as a result of that duplication we are adding further duplication of `CertificateGenerationHistory`. I think we should removed `generate_certificates_for_all_students` method and just use `generate_certificates_for_students`.